sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/CBDE/CheckMLIRWithReference.cs

```diff
@@ -302,7 +302,48 @@ private void g(int dim)
 ^bb2:	// pred: ^bb1
   return
 }
+";
+            MlirTestUtilities.ValidateWithReference(code, expected, TestContext.TestName);
+        }
 
+        [TestMethod]
+        public void Int32_Constant()
+        {
+            var code = @"
+void Func() {
+    const int Value = 42;
+    var a = int.MaxValue;
+    var b = int.MinValue;
+    var c = System.Int32.MaxValue;
+    var d = Value;
+}
+";
+            var expected = @"
+func @_$invalid$global$code$.Func$$() {
+  br ^bb1
+^bb1:	// pred: ^bb0
+  %c42_i32 = constant 42 : i32
+  %0 = cbde.alloca i32
+  cbde.store %c42_i32, %0 : memref<i32>
+  %1 = cbde.unknown : i32
```

Comment:
Does this line matches the `int.`, or the `MaxValue` part?
Anyways, it is useless (and could be removed), but it looks harmless...

Reply:
It matches the `int.` part. Cleaning this one would be another story (not too difficult). But it would not remove the `Int32.` variant (more difficult) so I've chosen to go with "harmless" approach.